internals/osutil/mkdirallchown_test.go

```diff
@@ -46,22 +48,131 @@ func (mkdacSuite) TestSlashySlashy(c *check.C) {
 func (mkdacSuite) TestMkdirAllChown(c *check.C) {
 	tmpDir := c.MkDir()
 
-	err := osutil.MkdirAllChown(tmpDir+"/foo/bar", 0o755, osutil.NoChown, osutil.NoChown)
+	err := osutil.MkdirAllChown(tmpDir+"/foo/bar", 0o755, 0, osutil.NoChown, osutil.NoChown)
 	c.Assert(err, check.IsNil)
 	c.Assert(osutil.IsDir(tmpDir+"/foo"), check.Equals, true)
 	c.Assert(osutil.IsDir(tmpDir+"/foo/bar"), check.Equals, true)
 
-	err = osutil.MkdirChown(tmpDir+"/foo/bar", 0o755, osutil.NoChown, osutil.NoChown)
+	err = osutil.MkdirChown(tmpDir+"/foo/bar", 0o755, 0, osutil.NoChown, osutil.NoChown)
 	c.Assert(err, check.ErrorMatches, `.*: file exists`)
 }
 
 func (mkdacSuite) TestMkdirChown(c *check.C) {
 	tmpDir := c.MkDir()
 
-	err := osutil.MkdirChown(tmpDir+"/foo", 0o755, osutil.NoChown, osutil.NoChown)
+	err := osutil.MkdirChown(tmpDir+"/foo", 0o755, 0, osutil.NoChown, osutil.NoChown)
 	c.Assert(err, check.IsNil)
 	c.Assert(osutil.IsDir(tmpDir+"/foo"), check.Equals, true)
 
-	err = osutil.MkdirChown(tmpDir+"/foo", 0o755, osutil.NoChown, osutil.NoChown)
+	err = osutil.MkdirChown(tmpDir+"/foo", 0o755, 0, osutil.NoChown, osutil.NoChown)
 	c.Assert(err, check.ErrorMatches, `.*: file exists`)
 }
+
+func (mkdacSuite) TestMkdirChownWithoutMkdirFlags(c *check.C) {
+	oldmask := syscall.Umask(0022)
+	defer syscall.Umask(oldmask)
+
+	tmpDir := c.MkDir()
+
+	err := osutil.MkdirChown(tmpDir+"/foo", 0o777, 0, osutil.NoChown, osutil.NoChown)
+	c.Assert(err, check.IsNil)
+	c.Assert(osutil.IsDir(tmpDir+"/foo"), check.Equals, true)
+
+	info, err := os.Stat(tmpDir + "/foo")
+	c.Assert(err, check.IsNil)
+	c.Assert(info.Mode().Perm(), check.Equals, os.FileMode(0o755))
+}
+
+func (mkdacSuite) TestMkdirChownWithMkdirFlags(c *check.C) {
+	tmpDir := c.MkDir()
+
+	err := osutil.MkdirChown(tmpDir+"/foo", 0o777, osutil.MkdirChmod, osutil.NoChown, osutil.NoChown)
+	c.Assert(err, check.IsNil)
+	c.Assert(osutil.IsDir(tmpDir+"/foo"), check.Equals, true)
+
+	info, err := os.Stat(tmpDir + "/foo")
+	c.Assert(err, check.IsNil)
+	c.Assert(info.Mode().Perm(), check.Equals, os.FileMode(0o777))
+}
+
+// The following test is commented out because it would fail in GitHub Actions, but it would be nice
```

Comment:
I don't have a good way to make it pass in GitHub Actions.

Unless I use the uid/gid 1001/127 under which user the workflow is running, but it seems the numbers might change (a few years back it was 1001/115), and it would probably fail locally.

Should I delete this test? But I feel it's mandatory to test this branch of the function.

Same below.

Reply:
Could we make this a "root test", like the others in `.github/workflows/tests.yaml` and use `PEBBLE_TEST_USER` and `PEBBLE_TEST_GROUP` like `TestMakeDirsUserGroupReal` does?